crates/bevy_ecs/src/reflect/entity_commands.rs

```diff
@@ -0,0 +1,451 @@
+        drop(registry);
+        world.insert_resource(type_registry);
+
+        let mut system_state: SystemState<Commands> = SystemState::new(&mut world);
+        let mut commands = system_state.get_mut(&mut world);
+
+        let entity = commands.spawn_empty().id();
+
+        let boxed_reflect_component_a = Box::new(ComponentA(916)) as Box<dyn Reflect>;
+
+        commands
+            .entity(entity)
+            .insert_reflect(boxed_reflect_component_a);
+        system_state.apply(&mut world);
+
+        assert!(world.entity(entity).get::<ComponentA>().is_some());
+        assert_eq!(world.entity(entity).get::<ComponentA>().unwrap().0, 916);
+    }
+
+    #[test]
+    fn insert_reflected_with_registry() {
+        let mut world = World::new();
+
+        let mut type_registry = TypeRegistryResource {
+            type_registry: TypeRegistry::new(),
+        };
+
+        type_registry.type_registry.register::<ComponentA>();
+        let registration = type_registry
+            .type_registry
+            .get_mut(std::any::TypeId::of::<ComponentA>())
+            .unwrap();
+        registration.insert(<ReflectComponent as FromType<ComponentA>>::from_type());
+        world.insert_resource(type_registry);
+
+        let mut system_state: SystemState<Commands> = SystemState::new(&mut world);
+        let mut commands = system_state.get_mut(&mut world);
+
+        let entity = commands.spawn_empty().id();
+
+        let boxed_reflect_component_a = Box::new(ComponentA(916)) as Box<dyn Reflect>;
+
+        commands
+            .entity(entity)
+            .insert_reflect_with_registry::<TypeRegistryResource>(boxed_reflect_component_a);
+        system_state.apply(&mut world);
+
+        assert!(world.entity(entity).get::<ComponentA>().is_some());
+        assert_eq!(world.entity(entity).get::<ComponentA>().unwrap().0, 916);
+    }
+
+    #[test]
+    fn remove_reflected() {
+        let mut world = World::new();
+
+        let type_registry = AppTypeRegistry::default();
+        let mut registry = type_registry.write();
+        registry.register::<ComponentA>();
+        let registration = registry
+            .get_mut(std::any::TypeId::of::<ComponentA>())
+            .unwrap();
+        registration.insert(<ReflectComponent as FromType<ComponentA>>::from_type());
+        drop(registry);
+        world.insert_resource(type_registry);
+
+        let mut system_state: SystemState<Commands> = SystemState::new(&mut world);
+        let mut commands = system_state.get_mut(&mut world);
+
+        let entity = commands.spawn(ComponentA(0)).id();
+
+        let boxed_reflect_component_a = Box::new(ComponentA(916)) as Box<dyn Reflect>;
+
+        commands
+            .entity(entity)
+            .remove_reflect(boxed_reflect_component_a.type_name().to_owned());
+        system_state.apply(&mut world);
+
+        assert!(world.entity(entity).get::<ComponentA>().is_none());
```

Comment:
```suggestion
        assert_eq!(world.entity(entity).get::<ComponentA>(), None);
```
This would improve the error message